### PR TITLE
Import more dtype aliases from NumPy

### PR DESCRIPTION
**Pull request by kmaehashi**

### CODE REVIEW

**Review by toslunar — APPROVED** [review]
LGTM

File: cupy/__init__.py [hunk 1]
```diff
@@ -398,6 +374,11 @@ def result_type(*arrays_and_dtypes):
 from numpy import mintypecode  # NOQA
 from numpy import sctype2char  # NOQA
 from numpy import typename  # NOQA
+from numpy import maximum_sctype  # NOQA
```
Comment (toslunar) [inline cupy/__init__.py:377]:
```
>>> numpy.maximum_sctype(float)
<class 'numpy.float128'>
```
in my environment. Should `cupy.maximum_sctype(float)` return `float64`?  (But it'll be surprising for `cupy.maximum_sctype(numpy.float128)` to return `float64`.)

Comment (kmaehashi) [reply cupy/__init__.py]: Thansk! Removed ones that cannot be imported directly and added to #6078.

**Review by toslunar — APPROVED** [review]
LGTM

The PR title would be "Import more dtype aliases from NumPy".

**Comment by kmaehashi** [comment]
Fixed!

**Pull request merged**